german-quinonez/lesson-1/homework-2-Car/src/Car.java

```diff
@@ -0,0 +1,73 @@
+public class Car {
+
+    private double gas;
+    private double distance;
+    private double gasConsume;
+    private boolean isPowerOn;
+    private double capacity;
+
+
+    public Car() {
+        this.gas = 0.0;
+        this.distance = 0.0;
+        this.gasConsume = 0.0;
+        this.isPowerOn = false;
+        this.capacity = 0.0;
+    }
+
+    public void powerOn() {
+        this.isPowerOn = true;
+    }
+
+    public void powerOff() {
+        this.isPowerOn = false;
+    }
+
+    public void fillGas(double gas) {
+        if (this.gas + gas <= this.capacity)
+            this.gas += gas;
+    }
+
+    private double maxDistance() {
+        return gasConsume * gas;
+    }
+
+    public void move(double distance) {
+        this.distance = maxDistance();
```

Comment:
This distance should be the distance the car traveled since it was created, so it should never be assigned to the maxDistance; it should only increase whenever the car moves

Reply:
fixed